plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -33,6 +34,68 @@
 
 logger = sgtk.LogManager.get_logger(__name__)
 
+class ProgressHandler(QtCore.QObject):
+    """
+    An object that wraps a QTimer that is used to periodically check
+    for updates that need to be made to the progress bar in Maya. This
+    will always execute progress updates on the main thread.
+    """
+    PROGRESS_INTERVAL = 150 # milliseconds
+
+    def __init__(self):
+        try:
```

Comment:
Why isn't this a top-level import? A comment would be appreciated if it can't be imported at the top of the file.